api/db-cdr/generate-cdr/make-bq-data.sh

```diff
@@ -177,9 +177,13 @@ echo "Inserting concept table data ... "
 bq --quiet --project=$BQ_PROJECT query --nouse_legacy_sql \
 "INSERT INTO \`$WORKBENCH_PROJECT.$WORKBENCH_DATASET.concept\`
 (concept_id, concept_name, domain_id, vocabulary_id, concept_class_id, standard_concept,
-concept_code, count_value, prevalence, source_count_value)
+concept_code, count_value, prevalence, source_count_value, synonyms)
+with concept_synonyms as
+(select c.concept_id as concept, concat(cast(c.concept_id as string),'|',c.concept_code,'|',c.concept_name,'|',string_agg(cs.concept_synonym_name,'|')) as synonyms
```

Comment:
We should consider escaping concept_name and concept_synonym_name so that | becomes ||. Otherwise if this character *does* appear in a name / synonym, we won't be able to parse it correctly later.

If we think we can pick some delimiter char that we're confident is never going to show up in OMOP names / synonyms, then I guess this isn't necessary.

Also: per discussion on Slack, we may want to add vocabulary_id here too.

Reply:
Escaped the pipe delimiter.

Reply:
So the one performance downside to include concept code and name in the synonyms column is it means if we read concept code and name and synonyms, we're getting the same data twice from the database... so slightly more memory overhead. (I'm not really worried about the disk.)

If we want to instead do what Peter suggests and add a full text index on name, code, synonyms, and update our JPA code to have match_concept() / match_3() or whatever, we'd avoid that. We should verify that searching across multiple columns performs the same and behaves the same way as if they were all mashed together.

I don't mind whether or not we try that in this PR. We could do that as a follow-up later. @biopete WDYT?

Reply:
We are seeing performance wise. Just taking a while for db to add ft indexes. Anyway, we should fix it in this pr. It will be fine. From my research nobody combines text fields in the same row just to index. It should not effect performance have 3 columns in the index. Much cleaner ..  Innodb indexes are done by tokens in separate index tables so basically every token is taken out of every field and put in an index table. So it doesn't matter what columns the tokens are in. 

I'm still puzzled why doing the join on the full text index to synonms takes so much longer than having synonms on concept ... I'm seeing if it is just a query thing.